lib/fetch.js

```diff
@@ -1,158 +1,185 @@
+    isSSLLoaded = true;
   }
+};
+
+/**
+ * Fetch whatever is at the provided URL. Use cached version if available.
+ * @param {*} url URL of the resource
+ * @param {*} type type of the resource
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {*} alwaysRun fetches from URL even if resource is in cache, defaults to false
+ */
+export const fetch = async (url, type, date = process.env.SCRAPE_DATE || datetime.getYYYYMD(), alwaysRun = false) => {
+  const cachedBody = await caching.getCachedFile(url, type, date);
+
+  if (cachedBody === caching.CACHE_MISS || alwaysRun) {
+    console.log('  🚦  Loading data for %s from server', url);
 
-  return body;
-}
+    // Add SSL certificates for sources that use non-standard ones
+    await loadSSLCerts();
 
-/*
-  Load the webpage at the given URL and return a Cheerio object
-*/
-async function page(url, date) {
-  const body = await fetch(url, 'html', date);
+    const response = await needle('get', url);
+    const fetchedBody = response.body.toString();
 
-  if (!body) {
+    await caching.saveFileToCache(url, type, date, fetchedBody);
+
+    return fetchedBody;
+  }
+  return cachedBody;
+};
+
+/**
+ * Load the webpage at the given URL and return a Cheerio object
+ * @param {*} url URL of the resource
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {*} alwaysRun fetches from URL even if resource is in cache, defaults to false
+ */
+export const page = async (url, date, alwaysRun = false) => {
+  const body = await fetch(url, 'html', date, alwaysRun);
+
+  if (body === undefined) {
     return null;
   }
-
   return cheerio.load(body);
-}
-
-/*
-  Load and parse JSON from the given URL
-*/
-async function json(url, date) {
-  const body = await fetch(url, 'json', date);
-
-  if (!body) {
+};
+
+/**
+ * Load and parse JSON from the given URL
+ * @param {*} url URL of the resource
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {*} alwaysRun fetches from URL even if resource is in cache, defaults to false
+ */
+export const json = async (url, date, alwaysRun = false) => {
+  const body = await fetch(url, 'json', date, alwaysRun);
+
+  if (body === undefined) {
     return null;
   }
-
   return JSON.parse(body);
-}
-
-/*
-  Load and parse CSV from the given URL
-*/
-function csv(url, date) {
+};
+
+/**
+ * Load and parse CSV from the given URL
+ * @param {*} url URL of the resource
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {*} alwaysRun fetches from URL even if resource is in cache, defaults to false
+ */
+export const csv = async (url, date, alwaysRun = false) => {
   return new Promise(async (resolve, reject) => {
-    const body = await fetch(url, 'csv', date);
+    const body = await fetch(url, 'csv', date, alwaysRun);
 
-    if (!body) {
+    if (body === undefined) {
```

Comment:
Hmmmm, I do we want to try to parse empty string and `null`? Same thing above too.